web_ui/frontend/app/registry/page.tsx

```diff
@@ -18,29 +18,112 @@
 
 "use client"
 
-import RateGraph from "@/components/graphs/RateGraph";
-import StatusBox from "@/components/StatusBox";
+import {Box, Button, Grid, Typography, Skeleton, Alert, Collapse} from "@mui/material";
+import React, {useEffect, useMemo, useState} from "react";
 
-import {TimeDuration} from "@/components/graphs/prometheus";
-
-import {Box, Grid, Typography} from "@mui/material";
-import FederationOverview from "@/components/FederationOverview";
-import {ServerTable} from "@/components/ServerTable";
-import NamespaceTable from "@/components/NamespaceTable";
+import {PendingCard, Card, NamespaceCardSkeleton, CreateNamespaceCard} from "@/components/Namespace";
+import Link from "next/link";
+import {Namespace, Alert as AlertType} from "@/components/Main";
+import UnauthenticatedContent from "@/components/layout/UnauthenticatedContent";
+import {Authenticated, getAuthenticated, isLoggedIn} from "@/helpers/login";
 
 
 export default function Home() {
 
+    const [data, setData] = useState<Namespace[] | undefined>(undefined);
+    const [alert, setAlert] = useState<AlertType | undefined>(undefined)
+    const [authenticated, setAuthenticated] = useState<Authenticated | undefined>(undefined)
+
+    const getData = async () => {
+
+        let data : Namespace[] = []
+
+        const url = new URL("/api/v1.0/registry_ui/namespaces", window.location.origin)
+
+        const response = await fetch(url)
+        if (response.ok) {
+            const responseData: Namespace[] = await response.json()
+            responseData.sort((a, b) => a.id > b.id ? 1 : -1)
+            responseData.forEach((namespace) => {
+                if (namespace.prefix.startsWith("/cache")) {
+                    namespace.type = "cache"
+                } else {
+                    namespace.type = "origin"
+                }
+            })
+            data = responseData
+        }
+
+        return data
+    }
+
+    const _setData = async () => {setData(await getData())}
+
+    useEffect(() => {
+        _setData();
+        (async () => {
+            if(await isLoggedIn()){
+                setAuthenticated(getAuthenticated() as Authenticated)
+            }
+        })();
+    }, [])
+
+    const pendingData = useMemo(
```

Comment:
Can we add some text under "Pending Registration" or "Namespace Registry" clarifying who the namespace listed below belong to? When I login as a regular user it's a bit confusing to see "Pending Registration" and "Origin" and "Cache" as I don't know if this is others' namespace or mine.

The help text can be based on the user role:
* "Namespace Registry" 
  * Admin users: Namespace registry shows all registered namespaces in the federation. As an admin user, you can check pending namespace registration and review them. Approved registration are listed under "Origin" or "Cache" table.
  * Regular users: Namespace registry shows your registered namespaces in the federation. You can edit your registrations prior to approval. If you want to edit your registration that has been approved, please contact your registry administrator.
* "Pending Registration"
  * Admin users: "Pending Registration are namespaces waiting for review and approve or deny. If a cache is in pending status, the director will not use that cache"
  * Regular users: ""

<img width="862" alt="Screenshot 2024-01-09 at 3 15 39 PM" src="https://github.com/PelicanPlatform/pelican/assets/41393704/a564d7ce-341e-4ffe-b03a-bf3247c3589c">

Reply:
Also, I think currently if I logged in as a regular user, I won't be able to see others namespace, right? It seems that I can only see the one I owned. We still want to show a list of _all_ namespaces in the federation somewhere in the UI.

Reply:
Ahh. Never mind. I think the regular user can still see _all_ namespace registrations (it was me who didn't register namespace under a different user and confused myself). Now the question is, how can a user tell if the namespace belongs to themselves?

Reply:
You can see all approved if you are not logged in.